apps/api/src/app/integrations/e2e/update-integration.e2e.ts

```diff
@@ -36,48 +37,126 @@ describe('Update Integration - /integrations/:integrationId (PUT)', function ()
+
+    const { body } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
+
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('The credentials are required to activate the integration');
+  });
+
+  it('should allow updating the integration with just identifier', async function () {
+    const integrationOne = await integrationRepository.create({
+      name: 'Test',
+      identifier: 'identifier',
+      providerId: EmailProviderIdEnum.SendGrid,
       channel: ChannelTypeEnum.EMAIL,
-      credentials: { apiKey: '123', secretKey: 'abc' },
       active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+
+    const payload = {
+      identifier: 'identifier2',
       check: false,
     };
 
-    // create integrations
-    await session.testAgent.post('/v1/integrations').send(firstProviderPayload);
-    const mailgunIntegrationId = (await session.testAgent.post('/v1/integrations').send(secondProviderPayload)).body
-      .data._id;
+    const {
+      body: { data },
+    } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
 
-    // create irrelevant channel -> should not be affected by the update
-    firstProviderPayload.channel = ChannelTypeEnum.SMS;
-    await session.testAgent.post('/v1/integrations').send(firstProviderPayload);
+    expect(data.identifier).to.eq(payload.identifier);
+  });
 
-    // update second integration
-    secondProviderPayload.active = true;
-    await session.testAgent.put(`/v1/integrations/${mailgunIntegrationId}`).send(secondProviderPayload);
+  it('should allow updating the integration with just name', async function () {
+    const integrationOne = await integrationRepository.create({
+      name: 'Test',
+      identifier: 'identifier',
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
 
-    const integrations = await integrationRepository.findByEnvironmentId(session.environment._id);
+    const payload = {
+      name: 'Test2',
+      check: false,
+    };
 
-    const firstProviderIntegration = integrations.find(
-      (i) => i.providerId.toString() === 'sendgrid' && i.channel.toString() === ChannelTypeEnum.EMAIL
-    );
-    const secondProviderIntegration = integrations.find((i) => i.providerId.toString() === 'mailgun');
-    const irrelevantProviderIntegration = integrations.find(
-      (i) => i.providerId.toString() === 'sendgrid' && i.channel.toString() === ChannelTypeEnum.SMS
-    );
+    const {
+      body: { data },
+    } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
+
+    expect(data.name).to.eq(payload.name);
+  });
+
+  it('should allow updating the integration with just environment', async function () {
```

Comment:
Can we add a quick test for when updating identifier and environment while in the other environment uses that identifier? Individually both behaviours are covered with this and a previous test, but (I haven't checked the internals yet) would be a nasty surprise if when together the checks wouldn't work or the order of the checks allowed to a database error. 😄